File: letter-grade/code/letter_grade.py
grade = [0.0, 0.0, 0.0, 0.0] # midterm, final, weighted avg and rounded
weight = [0.4, 0.6] # midterm and final exam weights
grade[0] = float(input('Enter the midterm exam grade: '))
grade[1] = float(input('Enter the final exam grade: '))
grade[2] = weight[0]*grade[0] + weight[1]*grade[1]
grade[3] = round(grade[2])
print("The (weighted) final grade is %.2f." % grade[2])
print("After rounding, the (weighted) final grade is %i." % int(grade[3]))
letterGrade = ["AA", "BA", "BB", "CB", "CC", "DC", "DD","FD", "FF"]
gradeTransition = [81, 76, 70, 60, 50, 45, 40, 30]
if grade[3] >= gradeTransition[0]: # AA
    print("You got %s in this course." % letterGrade[0])
elif grade[3] >= gradeTransition[1]: # BA
    print("You got %s in this course." % letterGrade[1])
elif grade[3] >= gradeTransition[2]: ## BB
    print("You got %s in this course." % letterGrade[2])
elif grade[3] >= gradeTransition[3]: # CB
    print("You got %s in this course." % letterGrade[3])
elif grade[3] >= gradeTransition[4]: # CC
    print("You got %s in this course." % letterGrade[4])
elif grade[3] >= gradeTransition[5]: #DC
    print("You got %s in this course." % letterGrade[5])
elif grade[3] >= gradeTransition[6]: # DD
    print("You got %s in this course." % letterGrade[6])
elif grade[3] >= gradeTransition[7]: # FD
    print("You got %s in this course" % letterGrade[7])
else: # FF
    print("You got %s in this course." % letterGrade[8])

File: letter-grade/code/letter_grade.m
grade = zeros(1,4); % midterm, final, weighted avg and rounded
weight = [0.4, 0.6]; % midterm and final exam weights
grade(1) = input('Enter the midterm exam grade: ');
grade(2) = input('Enter the final exam grade: ');
grade(3) = weight(1)*grade(1) + weight(2)*grade(2);
grade(4) = round(grade(3));
fprintf('The (weighted) final grade is %.2f.\n', grade(3));
fprintf('After rounding, the (weighted) final grade is %i.\n', grade(4));
letterGrade = {'AA', 'BA', 'BB', 'CB', 'CC', 'DC', 'DD', 'FD', 'FF'};
gradeTransition = [81, 76, 70, 60, 50, 45, 40, 30];
if (grade(3) >= gradeTransition(1)) % AA
    fprintf('You got %s in this course.\n', letterGrade{1,1});
else if (grade(3) >= gradeTransition(2)) % BA
        fprintf('You got %s in this course.\n', letterGrade{1,2});
    else if (grade(3) >= gradeTransition(3)) % BB
            fprintf('You got %s in this course.\n', letterGrade{1,3});
        else if (grade(3) >= gradeTransition(4)) % CB
                fprintf('You got %s in this course.\n', letterGrade{1,4});
            else if (grade(3) >= gradeTransition(5)) % CC
                    fprintf('You got %s in this course.\n', letterGrade{1,5});
                else if (grade(3) >= gradeTransition(6)) % DC
                        fprintf('You got %s in this course.\n', letterGrade{1,6});
                    else if (grade(3) >= gradeTransition(7)) % DD
                            fprintf('You got %s in this course.\n', letterGrade{1,7});
                        else if (grade(3) >= gradeTransition(8)) % FD
                                fprintf('You got %s in this course.\n', letterGrade{1,8});
                            else
                                fprintf('You got %s in this course.\n', letterGrade{1,9});
                            end
                        end
                    end
                end
            end
        end
    end
end